### PR TITLE
Fix OOM during image browsing

### LINKED ISSUE #5479: NC - Android client crashes when swiping images during full-img download
### Steps to reproduce
1. Open image folder and enlarge images. (Switch to full screen slide show)
2. Swipe left and right
3. If client offers to download original version of image as no smaller version is stored, do not answer, but swipe to another image. (Also had crashes w/o step 3)
4. Repeat a couple of times and client crashes

### Expected behaviour
Client should discard message, and display next image.

### Actual behaviour
Client crashes, displays "cause of error" and dies. Error report says "out of memory".
Ram never drops below 40% ~ 1.7 Gb free. (checked with RAM-monitoring tool)

### Environment data
Android version: 6.01

Device model: Lenovo YT3-X90F

Stock or customized system: stock

Nextcloud app version: 3.10.1

Nextcloud server version: 17.0.2

####  Client error report
Interesting (IMHO) information marked bold
************ CAUSE OF ERROR ************

java.lang.OutOfMemoryError: Failed to allocate a __44744716 byte__ allocation with __12068672 free bytes__ and 11MB until OOM
	at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
	at android.graphics.BitmapFactory.nativeDecodeStream(Native Method)
	at android.graphics.BitmapFactory.decodeStreamInternal(BitmapFactory.java:686)
	at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:633)
	at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:708)
	at com.owncloud.android.ui.adapter.DiskLruImageCache.getBitmap(DiskLruImageCache.java:121)
	at com.owncloud.android.datamodel.ThumbnailsCacheManager.getBitmapFromDiskCache(ThumbnailsCacheManager.java:225)
	at com.owncloud.android.ui.preview.PreviewImageFragment.onStart(PreviewImageFragment.java:254)
	at androidx.fragment.app.Fragment.performStart(Fragment.java:2632)
	at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:915)
	at androidx.fragment.app.FragmentManagerImpl.performPendingDeferredStart(FragmentManagerImpl.java:713)
	at androidx.fragment.app.FragmentManagerImpl.startPendingDeferredFragments(FragmentManagerImpl.java:1325)
	at androidx.fragment.app.FragmentManagerImpl.doPendingDeferredStart(FragmentManagerImpl.java:2178)
	at androidx.fragment.app.FragmentManagerImpl.execSingleAction(FragmentManagerImpl.java:1703)
	at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:299)
	at androidx.fragment.app.FragmentStatePagerAdapter.finishUpdate(FragmentStatePagerAdapter.java:259)
	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
	at androidx.viewpager.widget.ViewPager$3.run(ViewPager.java:273)
	at androidx.viewpager.widget.ViewPager.completeScroll(ViewPager.java:2005)
	at androidx.viewpager.widget.ViewPager.scrollToItem(ViewPager.java:692)
	at androidx.viewpager.widget.ViewPager.setCurrentItemInternal(ViewPager.java:670)
	at androidx.viewpager.widget.ViewPager.setCurrentItemInternal(ViewPager.java:631)
	at androidx.viewpager.widget.ViewPager.dataSetChanged(ViewPager.java:1086)
	at androidx.viewpager.widget.ViewPager$PagerObserver.onChanged(ViewPager.java:3097)
	at androidx.viewpager.widget.PagerAdapter.notifyDataSetChanged(PagerAdapter.java:291)
	at com.owncloud.android.ui.preview.PreviewImageActivity$DownloadFinishReceiver.onReceive(PreviewImageActivity.java:503)
	at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:881)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5609)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:746)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Lenovo
Device: YT3
Model: Lenovo YT3-X90F
Id: MMB29M
Product: YT3_10_row_wifi

************ FIRMWARE ************
SDK: 23
Release: 6.0.1
Incremental: YT3-X90F_USR_S200322_1706140050_WW17_BP_ROW

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix #5479

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/adapter/DiskLruImageCache.java [hunk 1]
```diff
@@ -116,9 +180,9 @@ public Bitmap getBitmap(String key) {
             }
             final InputStream in = snapshot.getInputStream(0);
             if (in != null) {
-                final BufferedInputStream buffIn =
-                        new BufferedInputStream(in, IO_BUFFER_SIZE);
+                final BufferedInputStream buffIn = new BufferedInputStream(in, IO_BUFFER_SIZE);
                 bitmap = BitmapFactory.decodeStream(buffIn);
+                buffIn.close();
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/adapter/DiskLruImageCache.java]: `InputStream in`doesn't get closed

closing `buffIn` and `in` should probably be handled in a finally block too (like above)

File: src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java [hunk 2]
```diff
@@ -306,6 +309,26 @@ public void onStart() {
         }
     }
 
+    private Bitmap getResizedBitmap(OCFile file, int width, int height) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java]: Annotate with `@Nullable` since return can be null?

**Comment by AndyScherzinger** [comment]
Just some minor comments

File: src/main/java/com/owncloud/android/ui/adapter/DiskLruImageCache.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/adapter/DiskLruImageCache.java:185]: Changed

File: src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java:312]: Totally true, we need to somehow enforce this.

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.java]: https://developer.android.com/studio/write/annotations#nullability-analysis
I'll give this a try.

**Comment by AndyScherzinger** [comment]
Spotbug score increased :/

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you rebase the PR please? ❤️

**Pull request merged**